### PR TITLE
LowRISC: Initial commit of Pad Controller

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

This PR adds support for the Pad Controller to the GPIO device.

### Testing Strategy

Booting Tock on the OT FPGA.

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

File: chips/lowrisc/src/gpio.rs [hunk 1]
```diff
@@ -104,7 +111,7 @@ impl GpioPin {
     }
 
     pub fn handle_interrupt(&self) {
-        let regs = self.registers;
+        let regs = self.gpio_registers;
```
Comment (ppannuto) [inline chips/lowrisc/src/gpio.rs:114]: Is there a reason for this renaming (and I mean the local `regs` variable from `self.gpio_registers`; the `registers` -> `gpio_registers` makes sense)?

I see how this is sort of copying what exists elsewhere in Tock, but I think that's more an accident of legacy MMIO design than intentional, and is hopefully on the way out (see #1973). I think while this is renaming this should just remove the `regs`.

Comment (alistair23) [reply chips/lowrisc/src/gpio.rs]: Removed

**Comment by ppannuto** [comment]
bors retry

**Pull request merged**